### PR TITLE
add lora example for clm and text generation

### PR DESCRIPTION
**Pull request by sywangyi**
# What does this PR do?

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests?

### CODE REVIEW

**Comment by sywangyi** [comment]
@libinta @yao-matrix please help review

**Review by regisss — COMMENTED** [review]
@sywangyi I left several comments but they are all minor and should be easily addressed.
I have not run the example yet, will do it to day or tomorrow and then we should be able to merge this!

File: examples/text-generation/run_generation.py [hunk 1]
```diff
@@ -242,6 +249,18 @@ def main():
     if not model.config.is_encoder_decoder:
         tokenizer.padding_side = "left"
     # Some models like GPT2 do not have a PAD token so we have to set it if necessary
+    if re.search("llama", model.config.architectures[0], re.IGNORECASE):
+        # unwind broken decapoda-research config
+        model.generation_config.pad_token_id = 0
+        model.generation_config.bos_token_id = 1
+        model.generation_config.eos_token_id = 2
```
[the remark this answers is not in the document]
Comment (sywangyi) [reply examples/text-generation/run_generation.py:256]: the defined number is incorrected. see https://huggingface.co/decapoda-research/llama-7b-hf/blob/main/generation_config.json. https://huggingface.co/decapoda-research/llama-7b-hf/blob/main/config.json
https://github.com/tloen/alpaca-lora/blob/main/generate.py#L76.   llama2 does not has such problem. see https://huggingface.co/meta-llama/Llama-2-7b-chat-hf/blob/main/generation_config.json

File: examples/language-modeling/run_lora_clm.py
[the remark this answers is not in the document]
Comment (sywangyi) [reply examples/language-modeling/run_lora_clm.py:1]: It's just modified from https://github.com/intel/intel-extension-for-transformers/blob/main/workflows/chatbot/fine_tuning/instruction_tuning_pipeline/finetune_clm.py. and we just add lora support to clm finetune.

**Review by regisss — COMMENTED** [review]
Thanks for the changes!

When trying the examples, I get:
```
ValueError: Tokenizer class LLaMATokenizer does not exist or is not currently imported.
```
which is due to a bad config file for `decapoda-research/llama-7b-hf`. Could you change to another model that is not affected by this issue?
use gpt2 instead, readme updated

**Comment by regisss** [comment]
@sywangyi I was rather talking about another Llama checkpoint that works properly. I just pushed a commit to update the examples. I'll merge once the CI passes.

**Review by regisss — APPROVED** [review]

**Pull request merged**